extension/src/arbitrary.rs

```diff
@@ -0,0 +1,53 @@
+
+use aggregate_builder::aggregate;
+
+use pgx::*;
+
+use crate::{
+    palloc::{Internal, InternalAsValue, Inner, ToInternal},
```

Comment:
The only thing I wonder is whether we can make these macros stand alone. It's very surprising that in order to use `#[aggregate]` I have to add a bunch of `use` that I'm not using, and presumably could grow or shrink with time!

I looked into it a bit and it seems at least some of it comes directly from pgx, but maybe that's something they'd accept?

Reply:
I have mixed feelings. On the one hand, I'd like this to work as much like an `impl` as possible, on the other hand I definitely agree that things that aren't visible shouldn't require `use`s. I think I'll see if I can have those use absolute paths...

Reply:
It looks pretty involved, anyway. I don't see any way to easily use these macros without leading with `use pgx::*` and a smattering of others.

We do have a couple stylistic issues we could address though:

- the outer layer of braces is unneeded:  `use crate::palloc::{Inner, Internal, InternalAsValue, ToInternal};`
- the two matches couple be replaced with `state.map(|s| s.into()).or(value)` and `state.map(|s| s.into())`, respectively. Yeah, this is only sample code, but sample code is expected to be copied and pasted, so arguably ought to be even nittier-pickier! :). Also clippy caught one of them and it would be nice to be clippy-clean.

Reply:
Removing the `use pgx::*` isn't going to be doable anytime soon; IIRC that requires some work in the expansion of `pg_extern`. The other ones however are unneeded as of the most recent commit! This allows things such as using `type AggregatePointer = Option<crate::palloc::Inner<AggState>>` and using `AggregatePointer` as the transition type, not that I'd recommend doing that.

> the two matches could be replaced with ... Also clippy caught one...
Ah, clippy. I actually disagree with that style; I think the explicit match is more readable, doesn't take much more space, and performs better (well, at compile time), without taking much more space; but the reason we added clippy to CI is to prevent style fights, so what it says goes. (In general this PR isn't clippy-clean yet, so I'd expect there to be a lot of such nits)